src/arrayMethodJoin.js

```diff
@@ -4,9 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
-  };
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += '';
+      } else {
```

Comment:
it's better to avoid nesting, you can use a conditional operator as an option https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator